LL-5352 Hardcode removal of Bitcoin dependency for Decred

juan-cortes opened the pull request:
## Context (issues, jira)
https://ledgerhq.atlassian.net/browse/LL-5352


## Description / Usage
There is an issue on backend side where Decred is listed as depending directly on the Bitcoin app, this means that we are currently blocking installations of the Decred app for users that don't have enough space to have the Bitcoin app too. This pr addresses the issue by hardcoding the removal of the dependency, but it's **worth noting that the fix should be done on backend** at the app store level and this piece of code should be removed once that's confirmed to be the case.

<!-- please share a sample of code that uses your new code (if features were added) -->
<!-- please document quickly what would the "user land" need to do to use the feature -->
![image](https://user-images.githubusercontent.com/4631227/131699967-2b1734d4-90a8-44e1-99fd-e0f13dffa3d5.png)
Only Decred without Bitcoin installed, correct storage size.

## Expectations

- [x] **Test coverage: The changes of this PR are covered by test.** Unit test were added with mocks when depending on a backend/device.
- [ ] **No impact: The changes of this PR have ZERO impact on the userland.** Meaning, we can use these changes without modifying LLD/LLM at all. It will be a "noop" and the maintainers will be able to bump it without changing anything.

<!--
If one of these can't be checked, please document it carefully (on the reason you can't check it).
NB: We want to avoid as much as possible such breaking changes to make a bump very easy.
-->

juan-cortes committed:
LL-5352 Hardcode removal of Bitcoin dependency for Decred

juan-cortes committed:
LL-5352 Test coverage

juan-cortes committed:
Merge branch 'master' into LL-5352

gre commented on src/apps/mock.ts:
```diff
@@ -92,7 +92,11 @@ export function mockListAppsResult(
-      const dependencies = getDependencies(name);
+      // FIXME LL-5352 until backend fixes the mistake, remove the dependencies for Decred.
+      const dependencies =
+        name === "Decred" || name === "Decred Testnet"
```
please put these in a whitelist object to avoid the double code.

juan-cortes replied:
Sure, I'll see what I can do boss

juan-cortes replied:
Putting comments allows the reviewer to understand the reasoning for the development, it's all part of the plan 🧠

juan-cortes committed:
Merge remote-tracking branch 'upstream/master' into LL-5352

juan-cortes committed:
Move whitelist dependencies to an independent const

Following the review, remove the comments hinting that backend had a bug when it doesnt, and clean up the code to avoid duplication.

juan-cortes committed:
Merge branch 'master' into LL-5352
